pkg/controller/installation/core_controller.go

```diff
@@ -963,6 +964,12 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		}
 	}
 
+	// Wait for IP pools to exist before continuing to install remaining components.
+	if len(instanceStatus.IPPools) == 0 {
```

Comment:
There's an argument to be made to switch this to just reading IPPools directly instead of waiting for the Installation status to be populated by the IP pools controller, and I feel like we should have that discussion.

I think the main point in favor of doing it this way is that it leaves us able to do whatever logic we want in the IP pools controller - validation, modification, etc - before telling this core controller that it's good to go and it allows us to do it all in one location. 

I also think that in terms of code readability, doing it this way makes it more clear that this controller is a dependent of the IP pool controller.

Adding this to the status also gives a central location to see all relevant IP pool information - the desired state as well as the actual cluster state - instead of needing to compare output from `kubectl get installation` and `kubectl get ippools`.

Using the IPPools in the data store directly means one less status field (a potentially large status field for clusters that have 100s of IP pools) and a slightly simpler to understand architecture with a single source of truth for which IP pools are actually installed in the cluster.

Reply:
I think I am actually changing my stance here and thinking we should just switch to reading these from the crd.projectcalico.org API instead of from the Installation status, mainly driven by thinking about really large clusters and how many IP pools may end up in the status.

Reply:
I've asked about using the IPPool resource in another comment. I think I like that better. The IPPools resource is the source of truth for what the cluster will be using, this status field would be as the ippool controllers sees things and not necessarily what the cluster is using. These will match +99% of the time so at that point I would ask why duplicate that information.